sorc/build_workflow_utils.sh

```diff
@@ -5,11 +5,11 @@ set -eux
 readonly UTILS_DIR=$(cd "$(dirname "$($cmd -f -n "${BASH_SOURCE[0]}" )" )" && pwd -P)
 
 # Adapt for global-workflow structure.
+source ${UTILS_DIR}/machine-setup.sh > /dev/null 2>&1
```

Comment:
can you move this line inside the if-block.
There is no such file when building on a generic platform.
This is specific to how we do things in our modulefiles.

Reply:
specifically after `set +x`.
The gibberish output from module commands makes no sense for the user to see.

Reply:
So our convention currently is not to provide $target as the build command is done (e.g. target=hera ./build_workflow_util.sh) but to provide it from the top of build_all.sh (source machine-setup.sh) and/or within the build script itself (which allows it to be run by itself outside of build_all.sh). Because of that we need $target set before the modulefile line otherwise the following happens and it never makes it into the if-block to run machine-setup and update $modulefile:

```
-bash-4.2$ sh build_workflow_utils.sh
++ uname -s
+ [[ Linux == Darwin ]]
++ which readlink
+ cmd=/usr/bin/readlink
++++ /usr/bin/readlink -f -n build_workflow_utils.sh
+++ dirname /scratch1/NCEPDEV/global/Kate.Friedman/git/feature-cmake_test/sorc/build_workflow_utils.sh
++ cd /scratch1/NCEPDEV/global/Kate.Friedman/git/feature-cmake_test/sorc
++ pwd -P
+ readonly UTILS_DIR=/scratch1/NCEPDEV/global/Kate.Friedman/git/feature-cmake_test/sorc
+ UTILS_DIR=/scratch1/NCEPDEV/global/Kate.Friedman/git/feature-cmake_test/sorc
+ target=NULL
+ modulefile=/scratch1/NCEPDEV/global/Kate.Friedman/git/feature-cmake_test/sorc/../modulefiles/workflow_utils.NULL
+ [[ -f /scratch1/NCEPDEV/global/Kate.Friedman/git/feature-cmake_test/sorc/../modulefiles/workflow_utils.NULL ]]
```

Reply:
You do set target in `build_all.sh`, why repeat that in every single build script?
You can do this:
target=$target ./build_workflow_utils.sh

Reply:
doing so, you are breaking the `build_workflow_utils.sh` for generic machines that don't have/need modules.

Reply:
> You can do this:
> target=$target ./build_workflow_utils.sh

That's perfectly fine for build_all.sh but then requires users to also provide $target like that when running that build script alone. That goes against the current convention...but I'll revert my change to build_workflow_utils.sh, add $target to build_all.sh for that build script and call it a day. Can come back to reworking the build scripts later.